### PR TITLE
Added section on minimum reqs issue to modify doc

### PR DESCRIPTION
**Pull request by Kamran64**
## Checklist

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Description
Outlines how users can set minimum requirements for their stack to be checked at init time. To be merged after the upcoming CLI release (and maybe after more work around the modify doc has been done)

## Related Issues
Related to: https://github.com/appsody/appsody/pull/535
#issue-number

### CODE REVIEW

File: content/docs/stacks/modify.md [hunk 1]
```diff
@@ -19,3 +19,24 @@ The `image` directory contains files for building the stack image. This image co
 The `templates` directory contains one or more starter applications that are given to the user when they initialise their projects. 
 
 Whilst modifying templates, consider if the functionality may be better placed in the stack image itself to affect all templates.
+
+## Advanced Topics
+
+### Setting stack requirements
+
+There may be cases where a modification you have made to a stack require the user to have a certain version of a technology associated with it e.g. Docker or the Appsody CLI. In these scenarios, it is important that you specify in the `stack.yaml` what these restrictions are to prevent the user from encountering errors further down the development line. Currently, the CLI supports enforcing restrictions on: Docker, Buildah, and the Appsody CLI.
```
Comment (helenmasters) [inline content/docs/stacks/modify.md]: I suggest rewording the following...

"You might modify a stack such that it requires the user to have a specific version, or range of versions, for a given technology. With the Appsody CLI, you can enforce version restrictions only when using Docker, Buildah, and the Appsody CLI."

Comment (helenmasters) [reply content/docs/stacks/modify.md]: If this is the first time we've referred to Buildah in this doc it should probably be a link.. but I don't know if that's the case yet.

File: content/docs/stacks/modify.md [hunk 2]
```diff
@@ -19,3 +19,24 @@ The `image` directory contains files for building the stack image. This image co
...
+version: 0.1.0
+.
+.
+.
+requirements:
+   docker-version: ">= 17.09.0"
+   appsody-version: "0.2.7 - 0.4.10"
+   buildah-version: "<= 1.10.0"
+```
+
```
Comment (helenmasters) [inline content/docs/stacks/modify.md]: I think we should have a sentence about what you can use here, something like:

"You can see the version comparisons that the Appsody CLI supports you using in `stack.yml` [here](https://github.com/Masterminds/semver#basic-comparisons)."

File: content/docs/stacks/modify.md [hunk 3]
```diff
@@ -19,3 +19,24 @@ The `image` directory contains files for building the stack image. This image co
...
+Following this, when the user intializes a project using the stack, the CLI will check the user matches the requirements set and if not, they will be unable to use the stack.
```
Comment (helenmasters) [inline content/docs/stacks/modify.md]: How about rewording to...

"When a project that uses the stack is initialized, the CLI checks whether the user meets the stack requirements. If the requirements are not met, the user cannot use the stack."

File: content/docs/stacks/modify.md (hunk 1 above)
Comment (Kamran64) [reply content/docs/stacks/modify.md]: I think the only other place that references Buildah is the CLI reference doc (flags on certain commands). Should this link to the Buildah GitHub page?

Comment (helenmasters) [reply content/docs/stacks/modify.md]: I'd link to the buildah.io page.

File: content/docs/stacks/modify.md [hunk 4]
```diff
@@ -24,7 +24,7 @@ Whilst modifying templates, consider if the functionality may be better placed i
 
 ### Setting stack requirements
 
-There may be cases where a modification you have made to a stack require the user to have a certain version of a technology associated with it e.g. Docker or the Appsody CLI. In these scenarios, it is important that you specify in the `stack.yaml` what these restrictions are to prevent the user from encountering errors further down the development line. Currently, the CLI supports enforcing restrictions on: Docker, Buildah, and the Appsody CLI.
+You might modify a stack such that it requires the user to have a specific version, or range of versions, for a given technology. With the Appsody CLI, you can enforce version restrictions only when using Docker, Buildah, and the Appsody CLI. In these scenarios, it is important that you specify in the `stack.yaml` what these restrictions are to prevent the user from encountering errors further down the development line.
```
Comment (helenmasters) [inline content/docs/stacks/modify.md]: "In these scenarios, it is important that you specify in the `stack.yaml` what these restrictions are to prevent the user from encountering errors further down the development line."

remove this line as it's not necessary
